<?xml version="1.0" encoding="UTF-8"?><!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml" lang="en"><head><meta http-equiv="Content-Type" content="text/html;charset=UTF-8"/><link rel="stylesheet" href="../../jacoco-resources/report.css" type="text/css"/><link rel="shortcut icon" href="../../jacoco-resources/report.gif" type="image/gif"/><title>RepositoryRuntimeExceptionMapper.java</title><link rel="stylesheet" href="../../jacoco-resources/prettify.css" type="text/css"/><script type="text/javascript" src="../../jacoco-resources/prettify.js"></script></head><body onload="window['PR_TAB_WIDTH']=4;prettyPrint()"><div class="breadcrumb" id="breadcrumb"><span class="info"><a href="../../jacoco-sessions.html" class="el_session">Sessions</a></span><a href="../../index.html" class="el_report">Fedora Repository HTTP API Module</a> &gt; <a href="../index.html" class="el_bundle">fcrepo-http-commons</a> &gt; <a href="index.source.html" class="el_package">org.fcrepo.http.commons.exceptionhandlers</a> &gt; <span class="el_source">RepositoryRuntimeExceptionMapper.java</span></div><h1>RepositoryRuntimeExceptionMapper.java</h1><pre class="source lang-java linenums">/*
 * The contents of this file are subject to the license and copyright
 * detailed in the LICENSE and NOTICE files at the root of the source
 * tree.
 */
package org.fcrepo.http.commons.exceptionhandlers;

import org.fcrepo.kernel.api.exception.RepositoryRuntimeException;
import org.slf4j.Logger;

import javax.ws.rs.core.Context;
import javax.ws.rs.core.Response;
import javax.ws.rs.ext.ExceptionMapper;
import javax.ws.rs.ext.Provider;
import javax.ws.rs.ext.Providers;

import static javax.ws.rs.core.Response.serverError;
import static org.fcrepo.http.commons.domain.RDFMediaType.TEXT_PLAIN_WITH_CHARSET;
import static org.slf4j.LoggerFactory.getLogger;

/**
 * @author cabeer
 * @since 9/13/14
 */

@Provider
public class RepositoryRuntimeExceptionMapper implements
        ExceptionMapper&lt;RepositoryRuntimeException&gt;, ExceptionDebugLogging {

    private final Providers providers;

    /**
     * Get the context Providers so we can rethrow the cause to an appropriate handler
     * @param providers the providers
     */
<span class="nc" id="L36">    public RepositoryRuntimeExceptionMapper(@Context final Providers providers) {</span>
<span class="nc" id="L37">        this.providers = providers;</span>
<span class="nc" id="L38">    }</span>

<span class="nc" id="L40">    private static final Logger LOGGER = getLogger(RepositoryExceptionMapper.class);</span>

    @Override
    public Response toResponse(final RepositoryRuntimeException e) {
<span class="nc" id="L44">        final Throwable cause = e.getCause();</span>
<span class="nc" id="L45">        ExceptionMapper&lt;Throwable&gt; exceptionMapper = null;</span>

<span class="nc bnc" id="L47" title="All 2 branches missed.">        if (cause != null) {</span>
<span class="nc" id="L48">            exceptionMapper = providers.getExceptionMapper((Class&lt;Throwable&gt;) cause.getClass());</span>
        }

<span class="nc bnc" id="L51" title="All 2 branches missed.">        if (exceptionMapper != null) {</span>
<span class="nc" id="L52">            return exceptionMapper.toResponse(cause);</span>
        }
<span class="nc" id="L54">        LOGGER.error(&quot;Caught a repository exception&quot;, e);</span>
<span class="nc" id="L55">        return serverError().entity(null).type(TEXT_PLAIN_WITH_CHARSET).build();</span>
    }
}
</pre><div class="footer"><span class="right">Created with <a href="http://www.jacoco.org/jacoco">JaCoCo</a> 0.8.7.202105040129</span></div></body></html>